[fusin02] iP

fusin02 opened the pull request:
# Dude

>“Your mind is for having ideas, not holding them.” – David Allen

Dude frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. Download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [ ] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders
- [ ] Many more
---
If you are a Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

fusin02 committed:
Renamed and implemented greet and exit

fusin02 committed:
Implemented Echo

fusin02 committed:
Implemented Add, List

fusin02 committed:
Implemented Mark as Done and created new class Task

fusin02 committed:
Implemented the 3 Tasks and handling edge cases for mark/unmark

fusin02 committed:
Edited test input and expected

fusin02 committed:
Added exception handling

fusin02 committed:
Added delete method and added exception handling for parseInt

fusin02 committed:
Added TaskType enum

fusin02 committed:
Added Javadocs

fusin02 committed:
Edit Javadocs

fusin02 committed:
Implement save data method

fusin02 committed:
Implement loading of data on start up

fusin02 committed:
Merge branch 'branch-Level-7' with load and save features

fusin02 committed:
Implement LocalDateTime feature

fusin02 committed:
Merge branch 'branch-Level-8' and implement DateTime

fusin02 committed:
Add more-OOP requirements

fusin02 committed:
Merge branch 'branch-A-MoreOOP'

fusin02 committed:
Remove unnecessary imports

fusin02 committed:
Fix naming for constants

fusin02 committed:
Implement packages

fusin02 committed:
Merge branch 'A-Gradle'

fusin02 committed:
Configure gradle

fusin02 committed:
Add JUnit tests

fusin02 committed:
Merge branch 'branch-A-JUnit' and implement JUnit tests

fusin02 committed:
Rename ShadowJar file

fusin02 committed:
Add JavaDocs for all public classes and methods

fusin02 committed:
Add Find feature

fusin02 committed:
Merge branch 'branch-A-JavaDoc' and implement JavaDoc

fusin02 committed:
Log commit

fusin02 committed:
Merge branch 'branch-A-CodingStandard'

fusin02 committed:
Merge branch-Level-9

fusin02 committed:
Remove unused import

fusin02 committed:
Rewire exception handling

fusin02 committed:
Add Checkstyle plugin

fusin02 committed:
Fix style errors from Checkstyle feedback

fusin02 committed:
Implement checkstyle

fusin02 committed:
Merge branch 'branch-A-CheckStyle'

fusin02 committed:
Add GUI

fusin02 committed:
Merge branch 'branch-Level-10'

Slipperybara commented on src/main/java/command/AddCommand.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    /**
+     * Adds the task to the task list and updates the storage file.
```
I like your documentation style

Slipperybara commented on src/main/java/command/AddCommand.java:
```diff
@@ -0,0 +1,48 @@
+/**
+ * Represents a command to add a task to the task list.
+ */
+public class AddCommand extends Command {
```
Class name should sound more like a noun, this naming make sense but it kind of sounds like a verb?

Slipperybara commented on src/main/java/exception/DudeException.java:
```diff
@@ -0,0 +1,15 @@
+/**
+ * Custom exception class for Dude-related exceptions.
+ */
+public class DudeException extends Exception {
```
I like how you name your exception, but maybe the naming could be abit more detailed?

Slipperybara commented on src/main/java/command/FindCommand.java:
```diff
@@ -0,0 +1,49 @@
+     * @throws DudeException If there is an error finding the tasks.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DudeException {
```
Maybe could add more detail to the naming of the execute method? Because every class have a execute method

Slipperybara reviewed (commented):
Looks good, some naming change requested

fusin02 committed:
Add varags

fusin02 committed:
Fix style issues

fusin02 committed:
Merge branch 'A-Varargs'

yuvrajaryan commented on src/main/java/command/AddCommand.java:
```diff
@@ -0,0 +1,48 @@
+     * @return false.
+     */
+    @Override
+    public boolean isExit() {
```
I like that you phrased the boolean method like a boolean

yuvrajaryan commented on src/main/java/dude/Dude.java:
```diff
@@ -0,0 +1,77 @@
+public class Dude {
+
+    private Storage storage;
+    private TaskList tasks;
```
I like that you named your tasklist as tasks.

yuvrajaryan commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,101 @@
+ * Represents a storage file that stores the tasks.
+ */
+public class Storage {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm");
```
I like that you used SCREAMING_SNAKE_CASE for your constants.

yuvrajaryan commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,44 @@
+    /**
+     * The end time of the event.
+     */
+    protected LocalDateTime to;
```
I think that you could add more detail to the variable name.

yuvrajaryan reviewed (commented):
LGTM

fusin02 committed:
Refactor name of Event variables

fusin02 committed:
Add Java assertions to ensure invariants at points

There are several invariants that should be in place
at parts of the code. Java assertions are put in place in those
lines.

This would add an additional layer of error handling and when
activated, will catch and stop the program when such invariants
no longer hold.

fusin02 committed:
Merge pull request #1 from fusin02/branch-A-Assertions

Add Java assertions to ensure invariants at points

fusin02 committed:
Refactor code to improve Code Quality

Check code for cases of poor code quality with respect to
course website. Added comments and abstracted methods to
improve code quality and readability.

fusin02 committed:
Merge pull request #2 from fusin02/branch-A-CodeQuality

Refactor code to improve Code Quality

fusin02 committed:
Add Streams usage in TaskList class

When using the Find Command, the TaskList's tasks are now converted
into a stream, filtered and recollected.

fusin02 committed:
Merge branch 'master' into branch-A-Streams

fusin02 committed:
Merge pull request #3 from fusin02/branch-A-Streams

Add Streams usage in TaskList class

fusin02 committed:
Fix style issues

fusin02 committed:
Add GitHub Actions for CI

fusin02 committed:
Fix style issues and update tests

fusin02 committed:
Remove IO Redirection Tests from GitHub Actions

fusin02 committed:
Add duplicate checking and more Java assertions

fusin02 committed:
Refactor code by extracting method out for readability

fusin02 committed:
Merge pull request #4 from fusin02/branch-BCD-Extension

Implement C-DetectDuplicates

fusin02 committed:
Add GUI functionality

fusin02 committed:
Rename Dude to Dynamike

fusin02 committed:
Update GUI to make it nicer

fusin02 committed:
Add personality to response

fusin02 committed:
Add Error Handling

fusin02 committed:
Add tests for delete command

fusin02 committed:
Fix checkstyle errors

fusin02 committed:
Fix CSS syntax error

fusin02 committed:
Add title to GUI

fusin02 committed:
Add Screenshot

fusin02 committed:
Improve Code Quality with SLAP

fusin02 committed:
Update README

fusin02 committed:
Add error handling for empty input
